crates/bevy_render/src/mesh/allocator.rs

```diff
@@ -0,0 +1,1025 @@
+                self.allocate(
+                    mesh_id,
+                    mesh.get_vertex_buffer_data().len() as u64,
+                    vertex_element_layout,
+                    &mut slabs_to_grow,
+                    mesh_allocator_settings,
+                );
+            } else {
+                self.allocate_large(mesh_id, vertex_element_layout);
+            }
+
+            // Allocate index data.
+            if let (Some(index_buffer_data), Some(index_element_layout)) =
+                (mesh.get_index_buffer_bytes(), ElementLayout::index(mesh))
+            {
+                self.allocate(
+                    mesh_id,
+                    index_buffer_data.len() as u64,
+                    index_element_layout,
+                    &mut slabs_to_grow,
+                    mesh_allocator_settings,
+                );
+            }
+        }
+
+        // Perform growth.
+        for (slab_id, slab_to_grow) in slabs_to_grow.0 {
+            self.reallocate_slab(render_device, slab_id, slab_to_grow);
+        }
+
+        // Copy new mesh data in.
+        for (mesh_id, mesh) in &extracted_meshes.extracted {
+            self.copy_mesh_vertex_data(mesh_id, mesh, render_device, render_queue);
+            self.copy_mesh_index_data(mesh_id, mesh, render_device, render_queue);
+        }
+    }
+
+    /// Copies vertex array data from a mesh into the appropriate spot in the
+    /// slab.
+    fn copy_mesh_vertex_data(
+        &mut self,
+        mesh_id: &AssetId<Mesh>,
+        mesh: &Mesh,
+        render_device: &RenderDevice,
+        render_queue: &RenderQueue,
+    ) {
+        let Some(&slab_id) = self.mesh_id_to_vertex_slab.get(mesh_id) else {
+            return;
+        };
+        let vertex_data = mesh.get_vertex_buffer_data();
+
+        // Call the generic function.
+        self.copy_mesh_data(
+            mesh_id,
+            mesh,
+            &vertex_data,
+            BufferUsages::VERTEX,
+            slab_id,
+            render_device,
+            render_queue,
+        );
+    }
+
+    /// Copies index array data from a mesh into the appropriate spot in the
+    /// slab.
+    fn copy_mesh_index_data(
+        &mut self,
+        mesh_id: &AssetId<Mesh>,
+        mesh: &Mesh,
+        render_device: &RenderDevice,
+        render_queue: &RenderQueue,
+    ) {
+        let Some(&slab_id) = self.mesh_id_to_index_slab.get(mesh_id) else {
+            return;
+        };
+        let Some(index_data) = mesh.get_index_buffer_bytes() else {
+            return;
+        };
+
+        // Call the generic function.
+        self.copy_mesh_data(
+            mesh_id,
+            mesh,
+            index_data,
+            BufferUsages::INDEX,
+            slab_id,
+            render_device,
+            render_queue,
+        );
+    }
+
+    /// A generic function that copies either vertex or index data into a slab.
+    #[allow(clippy::too_many_arguments)]
+    fn copy_mesh_data(
```

Comment:
```suggestion
    fn copy_element_data(
```

To match the terminology used elsewhere.

Reply:
Done